crates/fonticulus/src/basictables.rs

```diff
@@ -223,18 +224,24 @@ pub fn compile_os2(
         // sFamilyClass: info.open_type_os2_family_class... (not public)
         sFamilyClass: 0,
         panose: get_panose(info),
-        ulCodePageRange1: Some(0b01100000000000000000000110010011), // XXX
-        ulCodePageRange2: Some(0),                                  // XXX
-        ulUnicodeRange1: 0b10100001000000000000000011111111,        // XXX
-        ulUnicodeRange2: 0,                                         // XXX
-        ulUnicodeRange3: 0,                                         // XXX
-        ulUnicodeRange4: 0,                                         // XXX
+        ulCodePageRange1: Some(0),
+        ulCodePageRange2: Some(0),
+        ulUnicodeRange1: 0b10100001000000000000000011111111, // XXX
+        ulUnicodeRange2: 0,                                  // XXX
+        ulUnicodeRange3: 0,                                  // XXX
+        ulUnicodeRange4: 0,                                  // XXX
         usFirstCharIndex: *mapping.keys().min().unwrap_or(&0xFFFF) as u16,
         usLastCharIndex: *mapping.keys().max().unwrap_or(&0xFFFF) as u16,
         usLowerOpticalPointSize: None,
         usUpperOpticalPointSize: None,
         fsSelection: get_selection(info),
+    };
+    if info.open_type_os2_code_page_ranges.is_none() {
+        table.calc_code_page_ranges(&mapping);
+    } else {
+        table.int_list_to_code_page_ranges(info.open_type_os2_code_page_ranges.as_ref().unwrap());
```

Comment:
Thanks for your very detailed response.

I decided to convert the function into a method since I was planning on including a bin script which would recalculate the codepages for an existing font. I felt this was cleaner than the previous function implementation because I'd be duplicating a lot of the setup logic in the script. On the other hand, keeping it as a function means it isn't tied to a particular struct so we'll get improved reusability. I'm happy to revert back to a function based approach if Rune Bender etc will use these functions on sources.

Reply:
I'm happy to defer to your judgement on this, and we can always make changes down the road as our needs reveal themselves.